ff/src/biginteger/arithmetic.rs

```diff
@@ -8,13 +8,31 @@ macro_rules! adc {
     }};
 }
 
-/// Calculate a + b + carry, returning the sum and modifying the
-/// carry value.
+/// Sets a = a + b + carry, and returns the new carry.
 #[inline(always)]
-pub(crate) fn adc(a: u64, b: u64, carry: &mut u64) -> u64 {
-    let tmp = a as u128 + b as u128 + *carry as u128;
-    *carry = (tmp >> 64) as u64;
-    tmp as u64
+#[allow(unused_mut)]
+pub(crate) fn adc(a: &mut u64, b: u64, carry: u64) -> u64 {
+    let tmp = *a as u128 + b as u128 + carry as u128;
+    *a = tmp as u64;
+    (tmp >> 64) as u64
+}
+
+/// Sets a = a + b + carry, and returns the new carry.
```

Comment:
The main different is that this one has a smaller carry?

Reply:
Yes, same with the new `sbb_for_sub_with_borrow`